test/test_images/scheduler_target/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	"github.com/kelseyhightower/envconfig"
+)
+
+const (
+	eventSubject          = "subject"
+	eventData             = "data"
+	eventExtKnativecemode = "knativecemode"
+)
+
+func main() {
+	client, err := cloudevents.NewDefaultClient()
+	if err != nil {
+		panic(err)
+	}
+
+	r := Receiver{}
+	if err := envconfig.Process("", &r); err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("Waiting to receive event (timeout in %s seconds)...", r.Time)
+
+	duration, _ := strconv.Atoi(r.Time)
+	timer := time.NewTimer(time.Second * time.Duration(duration))
+	defer timer.Stop()
+
+	go func() {
+		<-timer.C
+		// Write the termination message if time out occured
+		fmt.Println("Timed out waiting for event from scheduler")
+		if err := r.writeTerminationMessage(map[string]interface{}{
+			"success": false,
+		}); err != nil {
+			fmt.Println("Failed to write termination message, got error:", err.Error())
+		}
+		os.Exit(0)
+	}()
+
+	if err := client.StartReceiver(context.Background(), r.Receive); err != nil {
+		log.Fatal(err)
+	}
+}
+
+type Receiver struct {
+	Time          string `envconfig:"TIME" required:"true"`
+	SubjectPrefix string `envconfig:"SUBJECT_PREFIX" required:"true"`
+	Knativecemode string `envconfig:"KNATIVECEMODE" required:"true"`
+	Data          string `envconfig:"DATA" required:"true"`
+}
+
+func (r *Receiver) Receive(event cloudevents.Event) {
+	// Print out event received to log
+	fmt.Printf("Received event\n")
+	fmt.Printf("	Context: %v\n", event.Context.String())
+	fmt.Printf("	Data: %s\n", event.Data)
+	fmt.Printf("	Encoded: %v\n", event.DataEncoded)
+	fmt.Printf("	Binary: %v\n", event.DataBinary)
+
+	incorrectAttributes := make(map[string]interface{})
+
+	// Check subject prefix
+	subject := event.Context.GetSubject()
+	if !strings.HasPrefix(subject, r.SubjectPrefix) {
+		incorrectAttributes[eventSubject] = subject
+	}
+
+	// Check knativecemode extension
+	if knativecemode, ok := event.Context.GetExtensions()["knativecemode"]; ok {
```

Comment:
`event.Context.GetExtension("knativecemode")`...  but I wouldn't care much about this knativecemode actually, that is something that we might remove anytime soon.
I'd rather check for the Source, Type and Subject, and the data if possible

Reply:
I can remove knativecemode